90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -424,6 +425,57 @@ select_kernel() {
   echo "${kexec_args}"
 }
 
+# arg1: ZFS filesystem
+# arg2: path for the mounted filesystem
+# prints: discovered prefix for root= command-line argument
+
+find_root_prefix() {
+  local zfsbe_mnt zfsbe_fs prefix
+  zfsbe_fs="${1}"
+  zfsbe_mnt="${2}"
+
+  # Grab the root prefix from a property if possible
+  if prefix="$( zfs get -H -o value org.zfsbootmenu:rootprefix "${zfsbe_fs}" )"; then
+    if [ "${prefix}" != "-" ]; then
+      echo "${prefix}"
+      return
+    fi
+  fi
+
+  # Try looking at os-release in BE
+  if [ -n "${zfsbe_mnt}" ]; then
+    prefix=$(
+      # OS type is in ID and ID_LIKE variables; /etc supersedes /usr/lib
+      unset ID ID_LIKE
+      for osrel in ${zfsbe_mnt}/{usr/lib,etc}/os-release; do
+        if [ -f "${osrel}" ]; then
+          # shellcheck disable=SC1090
+          . "${osrel}" >/dev/null 2>&1
+        fi
+      done
+
+      for ostype in $ID $ID_LIKE; do
+        case "$ostype" in
```

Comment:
Maybe add the Void and Ubuntu/Debian types here so they're explicitly documented. Ubuntu/Debian seem to be able to use the Dracut method of `zfs:`- https://github.com/openzfs/zfs/blob/master/contrib/initramfs/scripts/zfs#L758-L766

Reply:
That's resonable; done.